From 2c2cd6f26c458081496263516e6b2d63685c9b15 Mon Sep 17 00:00:00 2001
From: Jason Wang <jasowang@redhat.com>
Date: Mon, 29 Sep 2014 00:59:18 -0400
Subject: [virt] hyperv: NULL dereference on error

Message-id: <1411952358-34104-35-git-send-email-jasowang@redhat.com>
Patchwork-id: 96780
O-Subject: [RHEL7.1 PATCH 34/34] hyperv: NULL dereference on error
Bugzilla: 1146357
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Vitaly Kuznetsov <vkuznets@redhat.com>

From: Dan Carpenter <dan.carpenter@oracle.com>

Bugzilla: 1146357

We try to call free_netvsc_device(net_device) when "net_device" is NULL.
It leads to an Oops.

Fixes: f90251c8a6d0 ('hyperv: Increase the buffer length for netvsc_channel_cb()')
Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit b1c849276b5e199f94501101c07ee9cb6ced4935)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/hyperv/netvsc.c b/drivers/net/hyperv/netvsc.c
index 53a6ec3..cfdba85 100644
--- a/drivers/net/hyperv/netvsc.c
+++ b/drivers/net/hyperv/netvsc.c
@@ -1055,10 +1055,8 @@ int netvsc_device_add(struct hv_device *device, void *additional_info)
  struct net_device *ndev;
 
  net_device = alloc_net_device(device);
- if (!net_device) {
-  ret = -ENOMEM;
-  goto cleanup;
- }
+ if (!net_device)
+  return -ENOMEM;
 
  net_device->ring_size = ring_size;
 
-- 
1.7.1